cirq-google/cirq_google/ops/common_gate_families.py

```diff
@@ -0,0 +1,362 @@
+                f"be one of {_gate_list_to_str(self._supported_types.keys())}."
+            )
+
+        for g in gates_to_accept:
+            if isinstance(g, tuple(self._supported_types.keys())):
+                if cirq.is_parameterized(g):
+                    raise ValueError(
+                        f"Parameterized gate {g} cannot be used in `gates_to_accept` initializer."
+                    )
+            elif g not in self._supported_types:
+                raise ValueError(
+                    f"Gate {g} in `gates_to_accept` must be either a type from or an instance of "
+                    f"{_gate_list_to_str(self._supported_types.keys())}"
+                )
+        self.gates_to_accept = tuple(dict.fromkeys(gates_to_accept))
+        self.gate_types_to_check = tuple(dict.fromkeys(gate_types_to_check))
+        self.allow_symbols = allow_symbols
+        self.atol = atol
+        super().__init__(cirq.Gate)
+
+    def _default_name(self) -> str:
+        return f'FSimGateFamily: allow_symbol={self.allow_symbols}; atol={self.atol}'
+
+    def _default_description(self) -> str:
+        return (
+            f'`cirq_google.FSimGateFamily` which accepts any instance of gate types in'
+            f'\ngate_types_to_check: {_gate_list_to_str(self.gate_types_to_check)}'
+            f'\nwhich matches (across types), via instance check / value equality, a gate in'
+            f'\ngates_to_accept: {_gate_list_to_str(self.gates_to_accept)}'
+        )
+
+    def __repr__(self) -> str:
+        _gate_repr = lambda x: _gate_str(x, repr)
+        return (
+            'cirq_google.FSimGateFamily('
+            f'gates_to_accept={_gate_list_to_str(self.gates_to_accept, _gate_repr)},'
+            f'gate_types_to_check={_gate_list_to_str(self.gate_types_to_check, _gate_repr)},'
+            f'allow_symbols={self.allow_symbols},'
+            f'atol={self.atol})'
+        )
+
+    def _value_equality_values_(self) -> Any:
+        return (
+            frozenset(self.gates_to_accept),
+            frozenset(self.gate_types_to_check),
+            self.allow_symbols,
+            self.atol,
+        )
+
+    def _approx_eq_or_symbol(self, lhs: Any, rhs: Any) -> bool:
+        lhs = lhs if isinstance(lhs, tuple) else (lhs,)
+        rhs = rhs if isinstance(rhs, tuple) else (rhs,)
+        assert len(lhs) == len(rhs)
+        for l, r in zip(lhs, rhs):
+            is_parameterized = cirq.is_parameterized(l) or cirq.is_parameterized(r)
+            if (is_parameterized and not self.allow_symbols) or (
+                not is_parameterized and not cirq.approx_eq(l, r, atol=self.atol)
+            ):
+                return False
+        return True
+
+    def _get_value_equality_values(self, g: POSSIBLE_FSIM_GATES) -> Any:
+        # TODO: Remove condition once https://github.com/quantumlib/Cirq/issues/4585 is fixed.
+        if type(g) == cirq.PhasedISwapPowGate:
```

Comment:
nit: use isinstance (this should also allow removing the `type: ignore` on the next line)
```suggestion
        if isinstance(g, cirq.PhasedISwapPowGate):
```

Reply:
The reason to use type equality comparison  instead of `isinstance` was to ensure that we override this special case only for the class defined in Cirq and do not override the behavior for any user-defined derived class. So, a `CustomPhasedISwapGate` that derives from `PhasedISwapGate` and overrides the `_value_equality_values_` method wouldn't be special cased here.